jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/Content.java

```diff
@@ -193,7 +196,7 @@ static ByteBuffer asByteBuffer(Source source) throws IOException
          */
         static CompletableFuture<byte[]> asByteArrayAsync(Source source, int maxSize)
         {
-            return new ChunkAccumulator().readAll(source, maxSize);
+            return asRetainableByteBuffer(source, null, false, maxSize).thenApply(rbb -> BufferUtil.toArray(rbb.getByteBuffer()));
```

Comment:
This potentially performs multiple copies: both in `rbb.getByteBuffer()` and in `BufferUtil.toArray()`. Are we okay with this? And does this method have some value that we want to keep it around?

Reply:
Ah! I have a RBB.takeByteArray method already that will be better...